File: solutions/go/grains/1/grains.go
package grains

import (
	"fmt"
	"math"
)

// Square function returns the given square's number of grains.
func Square(n int) (uint64, error) {
	if !(1 <= n && n <= 64) {
		return 0, fmt.Errorf("invalid given square number")
	}
	return uint64(math.Exp2(float64(n - 1))), nil
}

// Total function returns the total number of grains on the board.
func Total() uint64 {
	var total, current uint64 = 0, 1
	for i := 0; i < 64; i++ {
		total += current
		current *= 2
	}
	return total
}

File: solutions/cpp/grains/1/grains.cpp
#include "grains.h"

namespace grains {

    std::uint64_t square(int num) {
        return 1ULL << (num - 1);
    }

    std::uint64_t total() {
        //return -1ULL; // :)
        return static_cast<std::uint64_t>(-1);
    }

}  // namespace grains
